proofs/tree.go

```diff
@@ -287,26 +303,95 @@ func NewSalt() (salt []byte) {
 // random values.
 //
 // This method will fail if there are any fields of type other than []byte (bytes in protobuf) in the
-// message.
-func FillSalts(message proto.Message) (err error) {
-	v := reflect.ValueOf(message).Elem()
+// message leaf.
+func FillSalts(dataMessage, saltsMessage proto.Message) (err error) {
+	dataMessageValue := reflect.Indirect(reflect.ValueOf(dataMessage))
+	saltsMessageValue := reflect.Indirect(reflect.ValueOf(saltsMessage))
+
+	for i := 0; i < saltsMessageValue.NumField(); i++ {
+		saltsField := saltsMessageValue.Field(i)
+		saltsType := reflect.TypeOf(saltsField.Interface())
+		valueField := dataMessageValue.FieldByName(saltsMessageValue.Type().Field(i).Name)
+
+		if saltsType.Kind() == reflect.Ptr {
+			saltsType = saltsType.Elem()
+		}
```

Comment:
I think we can use `reflect.Indirect(saltsType)` ?

Reply:
Reflect indirect works on `reflect.Value` not `reflect.Type`, and I wanted to have the checks on the type not `typeof(value.interface)`